src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserSessionPool.cs

```diff
@@ -190,16 +192,20 @@ internal void PutSession(TdsParserStateObject session)
             {
                 if (IsDisposed)
                 {
-                    // We're diposed - just clean out the session
+                    // We're disposed - just clean out the session
                     Debug.Assert(_cachedCount == 0, "SessionPool is disposed, but there are still sessions in the cache?");
                     session.Dispose();
                 }
                 else if ((okToReuse) && (_freeStateObjectCount < MaxInactiveCount))
                 {
                     // Session is good to re-use and our cache has space
                     SqlClientEventSource.Log.TryAdvancedTraceEvent("<sc.TdsParserSessionPool.PutSession|ADV> {0} keeping session {1} cachedCount={2}", ObjectID, session.ObjectID, _cachedCount);
+                    // TODO: To avoid merge conflict with TdsParserStateObject in PR# 1520, i.e. it'll be merged in the common code base, we can clean this up in a later PR.
+#if NETFRAMEWORK
                     Debug.Assert(!session._pendingData, "pending data on a pooled session?");
-
+#else
+                    Debug.Assert(!session.HasPendingData, "pending data on a pooled session?");
+#endif
```

Comment:
You can suggest the required changes and any possible conflict should be resolved in either #1520 or this PR based on the order of getting merged.